go.mod

```diff
@@ -1,6 +1,6 @@
 module go.etcd.io/etcd
 
-go 1.12
+go 1.16
```

Comment:
I think the idea was to keep this at 1.12 https://github.com/etcd-io/etcd/issues/12840#issuecomment-815534131 CC @ptabor

Reply:
No need to support go 1.12. It's out of support, and is subject to vulnerability. IIRC It is also not compatible with some system packages in newer version. 

Note that it's more than 1 year before when ptabor added that comment, and go 1.12 wasn't that old although it's also out of support at that time.

Reply:
I'm not 100% sure but I think go 1.16 keeps go compatibility [promise](https://go.dev/doc/go1.16), so its not subject to vulnerabilities. 
I also run into package issues but `go mod tidy` fixed it https://github.com/etcd-io/etcd/pull/14134/commits/ff898367250362288f5ce66fb7ed82be20210582

Either way is fine with me, but both  will work IMO.

Reply:
Note that the [promise of compatibility](https://go.dev/doc/go1compat) is for golang language of version < 2.0. There is no guarantee on the core library and tool-chain.